tests/test_settings.py

```diff
@@ -212,6 +214,22 @@ def test_check_settings_extra_check_parsers(self):
         ):
             DateDataParser(settings={'PARSERS': ['absolute-time', 'timestamp', 'absolute-time']})
 
+    def test_check_settings_extra_check_confidence_treshold(self):
+        with self.assertRaisesRegex(
+            SettingValidationError,
+            r'1.1 is not a valid value for '
+            r'LANGUAGE_DETECTION_CONFIDENCE_THRESHOLD. It can take values '
+            r'between 0 and 1'
+        ):
+            DateDataParser(settings={'LANGUAGE_DETECTION_CONFIDENCE_THRESHOLD': 1.1})
+
+    def test_check_settings_extra_check_default_languages(self):
+        with self.assertRaisesRegex(
+            SettingValidationError,
+            "Found invalid languages in the 'DEFAULT_LANGUAGES' setting: 'abcd'"
+        ):
+            DateDataParser(settings={'DEFAULT_LANGUAGES': ["abcd"]})
+
```

Comment:
I would add here a new test using the confidence threshold setting. You can do something like this:

```
def detect_languages(text, confidence_threshold):
     if confidence_threshold > 0.5:
         return ['en']
     else:
         return ['fr']

def test_confidence_threshold_setting_is_applied(self):
    ddp = DateDataParser(detect_languages_function=detect_languages, settings={'LANGUAGE_DETECTION_CONFIDENCE_THRESHOLD': 0.6})
    assert ddp.get_date_data('21/06/2020').locale == 'en'

    ddp2 = DateDataParser(detect_languages_function=detect_languages, settings={'LANGUAGE_DETECTION_CONFIDENCE_THRESHOLD': 0.4})
    assert ddp2.get_date_data('21/06/2020').locale == 'fr'

```

This is not exactly a unit test, but we try to ensure that the LANGUAGE_DETECTION_CONFIDENCE_THRESHOLD setting correctly arrives to the `datect_languages_function`.

Reply:
Thanks settings added